src/agent/communicator/src/communicator.cpp

```diff
@@ -111,7 +111,8 @@ namespace communicator
                 try
                 {
                     const auto result = SendAuthenticationRequest();
-                    if (result != boost::beast::http::status::ok)
+                    if (result < boost::beast::http::status::ok ||
```

Comment:
I think this is not necessary, function `SendAuthenticationRequest` always returns ok.